### ISSUE TITLE
Repo donation: Pluies/zeitgeist --> Release Engineering

### ISSUE THREAD

**Issue by justaugustus** [kubernetes/org#2172]
### New Repo, Staging Repo, or migrate existing
https://github.com/Pluies/zeitgeist

### Requested name for new repository
zeitgeist

### Which Organization should it reside
kubernetes-sigs

### If not a staging repo, who should have admin access
@kubernetes/sig-release-admins 

### If not a staging repo, who should have write access
@kubernetes/sig-release-admins + @Pluies

### If not a staging repo, who should be listed as approvers in OWNERS
Covered in https://github.com/Pluies/zeitgeist/pull/7

### If not a staging repo, who should be listed in SECURITY_CONTACTS
Covered in https://github.com/Pluies/zeitgeist/pull/7

### What should the repo description be
Zeitgeist: the language-agnostic dependency checker

### What SIG and subproject does this fall under in sigs.yaml
SIG Release / Release Engineering subproject

### Approvals

Approving as a SIG Release Chair after discussing donation with @Pluies.
Adding other @kubernetes/sig-release-leads for approvals.

### Additional context for request

Zeitgeist was created by @Pluies and is inspired by @yastij's [script to manage external dependencies](https://groups.google.com/forum/?pli=1#!topic/kubernetes-dev/cTaYyb1a18I) and extended to include checking with upstream sources to ensure dependencies are up-to-date.

@yastij -- let us know if you'd be interested in helping maintain as well!

### TODO

- [x] (@Pluies) Merge https://github.com/Pluies/zeitgeist/pull/7 
- [x] (@Pluies) Add boilerplate header text to all files
- [x] (@Pluies) Sign the CLA

cc: @kubernetes/release-engineering 
/sig release
/area release-eng dependency
Slack tracking: https://kubernetes.slack.com/archives/C2C40FMNF/p1599121368081100

**Comment by justaugustus** [comment]
cc: @kubernetes/owners 
/assign @nikhita

**Comment by nikhita** [comment]
Needs the following things before we can transfer the repo:

- The Kubernetes boilerplate header needs to be added to all files - https://github.com/kubernetes/kubernetes/tree/master/hack/boilerplate

- Template files like code-of-conduct, etc need to be added. Also, needs addition of this section - https://github.com/kubernetes/kubernetes-template-project#community-discussion-contribution-and-support to the README. All template files can be found at https://github.com/kubernetes/kubernetes-template-project.

**Comment by nikhita** [comment]
Ah, just saw that https://github.com/Pluies/zeitgeist/pull/7 addressed the second point. Thanks @justaugustus!

**Comment by justaugustus** [comment]
>     * The Kubernetes boilerplate header needs to be added to all files - https://github.com/kubernetes/kubernetes/tree/master/hack/boilerplate

@nikhita -- I purposely skipped this part, as our docs mention this can be added shortly after migration. More so, legally, I wasn't sure if that needed to be added by the repo owner. If it's actually cool for me to add it, I can update the PR with the boilerplate text.

**Comment by nikhita** [comment]
> I purposely skipped this part, as our docs mention this can be added shortly after migration.

Yes, we can technically do this after migration as well. Ideally, we try to get it in before migration to ensure all boxes are ticked though. :woman_shrugging: 

> More so, legally, I wasn't sure if that needed to be added by the repo owner. 

You are right! It needs to be added by the repo owner from a legal standpoint. In this case, @pluies would need to add them.

**Comment by justaugustus** [comment]
@Pleuis -- you'll also need to sign the CLA (Contributor License Agreement).
Details here: https://github.com/kubernetes/community/blob/master/CLA.md#how-do-i-sign

Let us know if you have any questions! :)

**Comment by justaugustus** [comment]
> You are right! It needs to be added by the repo owner from a legal standpoint. In this case, @Pluies would need to add them.

SGTM! Added a TODO section for Florent in the issue description.

**Comment by Pluies** [comment]
Thank you both for the guidance – here is the PR adding the boilerplate for review :) https://github.com/Pluies/zeitgeist/pull/8

**Comment by Pluies** [comment]
I went to sign the CLA and realised I had already signed it in February 2019, so it should be all good! 🤞

**Comment by justaugustus** [comment]
@nikhita -- it sounds like we've got everything we need from Florent. What's up next?

**Comment by nikhita** [comment]
@Pluies since this is a user-owned repo vs an org-owned repo, we'll need to first transfer the repo to a dummy org for moving to k-sigs (GitHub doesn't allow adding admins to user-owned repos).

Can you move the repo to https://github.com/kubernetes-purgatory when you get a chance? I can then move it to @kubernetes-sigs.

**Comment by Pluies** [comment]
Hi @nikhita ,
All done! Zeitgeist is now in https://github.com/kubernetes-purgatory/zeitgeist :)

**Comment by nikhita** [comment]
Repo has been transferred - https://github.com/kubernetes-sigs/zeitgeist :tada: 

Created:

- https://github.com/kubernetes/org/pull/2179 - to add GitHub teams to grant access
- https://github.com/kubernetes/community/pull/5104 - to add the repo to sigs.yaml

Once both PRs merge, we can close this issue.

---

@Pluies in order to get write access on the repo, you'll need to apply for membership to the @kubernetes-sigs org. https://github.com/kubernetes/community/blob/master/community-membership.md#member has instructions on how to apply. :)

**Comment by yastij** [comment]
@Pluies @justaugustus - happy to help maintain it !

**Comment by nikhita** [comment]
Teams have been granted access and all PRs have merged. Closing!

**Comment by justaugustus** [comment]
Thanks everyone. You all rock!

**Comment by justaugustus** [comment]
Follow-up PR adding @Pluies and @yastij is here: https://github.com/kubernetes/org/pull/2180

**Comment by Pluies** [comment]
Thanks for getting this done! 🙏 

> @Pluies in order to get write access on the repo, you'll need to apply for membership to the @kubernetes-sigs org. https://github.com/kubernetes/community/blob/master/community-membership.md#member has instructions on how to apply. :)

I think I've got everything on the checklist done... Except for the sponsors! And that should be arranged before opening the PR, so – @justaugustus and @nikhita , would you be happy to be sponsoring my application by any chance? :)

**Comment by nikhita** [comment]
Yes, feel free to count me in! :)

**Comment by justaugustus** [comment]
@Pluies -- you can go ahead and file a [membership request for kubernetes-sigs](https://github.com/kubernetes/org/issues/new?assignees=&labels=area%2Fgithub-membership&template=membership.md&title=REQUEST%3A+New+membership+for+%3Cyour-GH-handle%3E) and we can sort membership there.

Nikhita, @tpepper, and I are all at VMware, so you'll need one more non-VMware sponsor, which I'm sure either of the other SIG Release leads (@saschagrunert, @alejandrox1) will be happy to provide. :)

**Comment by saschagrunert** [comment]
Yes, happy to sponsor as well. 👍

**Comment by Pluies** [comment]
Thank you @nikhita @justaugustus @saschagrunert - PR open at https://github.com/kubernetes/org/issues/2181 :)
